### ISSUE TITLE
Mod Mail

### ISSUE THREAD

**Issue by Archasion** [iamysko/cds#55]
## Problem
Currently, dealing with reports is a bit tedious since most of the time users don't provide any proof for reports such as DM advertising, phishing and etc. Each user needs to be DMd separately and be asked to provide evidence for further decisions to be made.

## Solution
Users will be able to DM the bot to create a ticket which will be added to the queue (the bot should ask for confirmation that the user wants the ticket to be created via a reply or a reaction). After a ticket has been created it will be sent to the `#ticket-pool`/`#ticket-queue` and the user will receive an embed (in DMs) with some answers to common issues (E.g. Contact support if you've been hacked, report underage users to Discord's T&S, etc), if the user found the solution to their problem they will be able to say `cancel` or add a reaction to the embed to remove their ticket from the queue.

Once a ticket is 10th (or lower) in the queue, a channel should be created so a moderator can review the ticket and respond to it. Once the ticket has been solved the channel will be deleted and the conversation will be logged.

## Alternative Method
A `#ticket-pool`/`#ticket-queue` will be created in which a moderator can review the ticket, react to it to claim it and proceed to review the issue further in the bot's DMs.

### Pros:
- Only one channel needs to be created, the ticket queue.

### Cons:
- Pretty much exactly the same as the `#report-queue` except the moderator DMs the bot instead of the user.

## TL;DR
A ticket system that will let users' reports be dealt with more efficiently.
